w3af/plugins/grep/vulners_scanner.py

```diff
@@ -0,0 +1,251 @@
+"""
+vulners_scanner.py
+
+Copyright 2018 Vulners.com Team: Kir Ermakov (isox@vulners.com)
+
+This file is part of w3af, http://w3af.org/ .
+
+w3af is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation version 2 of the License.
+
+w3af is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with w3af; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+"""
+
+# w3af object imports here
+import w3af.core.data.constants.severity as severity
+from w3af.core.controllers.plugins.grep_plugin import GrepPlugin
+from w3af.core.data.bloomfilter.scalable_bloom import ScalableBloomFilter
+from w3af.core.data.quick_match.multi_re import MultiRE
+from w3af.core.data.kb.vuln import Vuln
+import w3af.core.controllers.output_manager as om
+from w3af.core.data.options.opt_factory import opt_factory
+from w3af.core.data.options.option_types import STRING
+from w3af.core.data.options.option_list import OptionList
+from w3af.core.data.kb.info_set import InfoSet
+from w3af.core.data.parsers.doc.url import URL
+# Additional modules imports
+import vulners
+import collections
+import re
+import json
+
+class vulners_scanner(GrepPlugin):
+    """
+    Find software vulnerabilities using Vulners.com API
+
+    Tests from the console mode:
+
+    With API key:
+
+    plugins grep vulners_scanner
+    plugins grep config vulners_scanner
+    set vulners_api_key YOUR_API_KEY_HERE
+    back
+    target set target VULNERABLE_WEBSITE
+    start
+
+
+    Without API key:
+
+    plugins grep vulners_scanner
+    target set target VULNERABLE_WEBSITE
+    start
+
+    :author: Vulners.com Team: Kir Ermakov (isox@vulners.com)
+    """
+
+    def __init__(self):
+        GrepPlugin.__init__(self)
+        # Vulners Rules JSON url
+        self._vulners_rules_url = URL('https://raw.githubusercontent.com/vulnersCom/detect-rules/master/rules.json')
+
+        # Vulners shared objects
+        self._vulners_api = None
+        self._vulners_api_key = None
+        self.rules_table = None
+        self.rules_updated = False
```

Comment:
`self._multi_re` should be defined as None here too.

Reply:
Got it.